src/backend/distributed/commands/statistics.c

```diff
@@ -467,19 +467,22 @@ GetExplicitStatisticsCommandList(Oid relationId)
 		}
 	}
 
-	/* find indexes on current relation, in case of modified stats target */
-	Relation relation = relation_open(relationId, AccessShareLock);
-	List *indexOidList = RelationGetIndexList(relation);
-	relation_close(relation, NoLock);
-
-	Oid indexId = InvalidOid;
-	foreach_oid(indexId, indexOidList)
+	if (includeIndexes)
 	{
-		/* we need alter index commands for altered targets on expression indexes */
-		List *alterIndexStatisticsCommands = GetAlterIndexStatisticsCommands(indexId);
+		/* find indexes on current relation, in case of modified stats target */
+		Relation relation = relation_open(relationId, AccessShareLock);
```

Comment:
hmz, something seems a little off here. This probably shouldn't have been in GetExplicitStatisticsCommandList to begin with.